### DISCUSSION TITLE
Field

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#35] [Feedback]
Use this place to discuss the [field component](https://ons-design-system.netlify.com/components/field) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by dudelmeister** [comment]
For some reason fields with a prefix or suffix are showing a height of 41px, single line input fields should have a consistent height of 40px (isn't this normally inherited?)

**Comment by jrbarnes9** [comment]
> For some reason fields with a prefix or suffix are showing a height of 41px, single line input fields should have a consistent height of 40px (isn't this normally inherited?)

Resolved in #117

**Comment by bameyrick** [comment]
The `legend` field should accept HTML for when the legend is also the `<h1>` of the page.

**Comment by bameyrick** [comment]
When a field is called by another component (e.g. `onsInput`, `onsSelect`, `onsCheckboxes`, `onsRadios`, `onsDateInput`, `onsPassword`, `onsTextarea`,  `onsUpload`) it should be possible to set field classes, field ids, field attributes, and legend classes from the calling component.

**Comment by bameyrick** [comment]
> The `legend` field should accept HTML for when the legend is also the `<h1>` of the page.

Resolved in #159

**Comment by bameyrick** [comment]
> When a field is called by another component (e.g. `onsInput`, `onsSelect`, `onsCheckboxes`, `onsRadios`, `onsDateInput`, `onsPassword`, `onsTextarea`, `onsUpload`) it should be possible to set field classes, field ids, field attributes, and legend classes from the calling component.

Resolved in #161

**Comment by bameyrick** [comment]
The last `field__item` or the last nested field shouldn't have a `margin-bottom` if it is not wrapped in `field__group`.

Also, `field__group` should have a `margin-bottom` of -1 * the `margin-bottom` of the inner `field__items`.

**Comment by bameyrick** [comment]
Having looked at the layout of a question form, and following discussions with @will-madebymoon and @jrbarnes9, we should probably move the styling for spacing between fields to the fields themselves rather than from `_question.scss`.

**Comment by bameyrick** [comment]
We should probably to update the field to component to use `<p>` tags (when not rendering as a `<fieldset>`) as recommended by WHATWG: https://html.spec.whatwg.org/multipage/forms.html#writing-a-form's-user-interface

**Comment by bameyrick** [comment]
The noField parameter is not documented on any inputs that utilise the field component. We should document this on all inputs.

**Comment by bameyrick** [comment]
When a field has a mutually exclusive option the labels are not rendering.

**Comment by bameyrick** [comment]
The `legend` has `display: table` set by default, which may prevent screen readers from reading it when they're not visually hidden.

**Comment by bameyrick** [comment]
Using to automatically switch between fieldsets, and p tags based on whether they are calling input has `mutuallyExclusive`(#29) config specified is too complex and causes issues when in use in projects like runner. We should separate all the logic held in the field component into separate mutually exclusive, field, and fieldset macros that are simpler to use.

**Comment by bameyrick** [comment]
Field groups don't space themselves out correctly:

![Screenshot 2019-07-09 at 15 23 00](https://user-images.githubusercontent.com/8447446/60896003-7760c680-a25d-11e9-9e73-9742d3a8fd29.png)

**Comment by bameyrick** [comment]
Input does not render in eq-survey runner with mutually exclusive set as the wrapping onsField element has no params object set.

**Comment by jrbarnes9** [comment]
Incorrect description of field parameter 'inline'.

It adds a class `ons-field--inline` to the field `div`, which displays the label inline with the input. It doesn't allow users to place multipe fields side-by-side.

**Reply by jrbarnes9** [reply]
Raised #2568
